src/components/pages/Groups/Groups.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+  ModularTable,
+  Spinner,
+  Button,
+  ContextualMenu,
+  Icon,
+  ButtonAppearance,
+  CheckboxInput,
+} from "@canonical/react-components";
+import type { ReactNode } from "react";
+import { useMemo, type JSX } from "react";
+
+import type { Group } from "api/api.schemas";
+import { useGetGroups } from "api/groups/groups";
+import Content from "components/Content";
+import ErrorNotification from "components/ErrorNotification";
+import NoEntityCard from "components/NoEntityCard";
+import { useEntitiesSelect, usePanel } from "hooks";
+import { Endpoint } from "types/api";
+
+import { Label } from "./types";
+
+const COLUMN_DATA = [
+  {
+    Header: "group name",
+    accessor: "groupName",
+  },
+  {
+    Header: "actions",
+    accessor: "actions",
+  },
+];
+
+const Groups = () => {
+  const { data, isFetching, isError, error, refetch } = useGetGroups();
+  const { generatePanel, openPanel, isPanelOpen } = usePanel<{
+    editGroupId?: string | null;
+    deleteGroups?: Group["name"][];
+    // TODO: display panels.
+  }>(() => null);
+
+  const {
+    handleSelectEntity: handleSelectGroup,
+    handleSelectAllEntities: handleSelectAllGroups,
+    selectedEntities: selectedGroups,
+    areAllEntitiesSelected: areAllGroupsSelected,
+  } = useEntitiesSelect(data?.data.data.map(({ name }) => name) ?? []);
+
+  const tableData = useMemo<Record<string, ReactNode>[]>(() => {
+    const groups = data?.data.data.map(({ name }) => name);
```

Comment:
Other than not bringing over the panels, these two maps are the only change to this component (in the feature branch the data is already a list of names).